Added base for regional/mega form pokdex data

TheXaman opened the pull request:
## Description
Alolan data from bulbapedia and galarian from psf#2936.

Added new function to retrieve the correct weight/height for forms, so that weight based attack moves correctly calculate

Draft PR for now, pokedex entries have to decided: https://pastebin.com/0M2Gk4Ti

## **Discord contact info**
TheXaman#5612

LOuroboros commented on src/pokedex.c:
```diff
@@ -4249,6 +4249,24 @@ u16 GetPokedexHeightWeight(u16 dexNum, u8 data)
+            return 1;
+        }
+    }
+    else
+        return GetPokedexHeightWeight(SpeciesToNationalPokedexNum(species), data);
```
```suggestion
    else
    {
        return GetPokedexHeightWeight(SpeciesToNationalPokedexNum(species), data);
    }
```

TheXaman commented:
Added all 28 megas and 2 primals based on the data from https://pokemondb.net/pokedex/stats/height-weight

AsparagusEduardo commented:
Just remembered that we're also gonna need category, seeing as Galarian Ponyta and Rapidash have different ones from Kantonian

equezada-jej commented on src/battle_script_commands.c:
```diff
@@ -9341,7 +9341,7 @@ static void Cmd_weightdamagecalculation(void)
     for (i = 0; sWeightToDamageTable[i] != 0xFFFF; i += 2)
     {
-        if (sWeightToDamageTable[i] > GetPokedexHeightWeight(SpeciesToNationalPokedexNum(gBattleMons[gBattlerTarget].species), 1))
+        if (sWeightToDamageTable[i] > GetSpeciesHeightWeight(gBattleMons[gBattlerTarget].species, 1))
```
I think that instead of creating a new function for this, we should modify `GetPokedexHeightWeight` instead, as modifying `battle_script_commands.c` can potentially create a conflict with BE.

AsparagusEduardo commented on src/battle_script_commands.c:
```diff
@@ -9341,7 +9341,7 @@ static void Cmd_weightdamagecalculation(void)
     for (i = 0; sWeightToDamageTable[i] != 0xFFFF; i += 2)
     {
-        if (sWeightToDamageTable[i] > GetPokedexHeightWeight(SpeciesToNationalPokedexNum(gBattleMons[gBattlerTarget].species), 1))
+        if (sWeightToDamageTable[i] > GetSpeciesHeightWeight(gBattleMons[gBattlerTarget].species, 1))
```
I think that instead of creating a new function for this, we should modify GetPokedexHeightWeight instead, as modifying battle_script_commands.c can potentially create a conflict with BE.

TheXaman replied:
if we modify `GetPokedexHeightWeight`we would have to change its input argument to species, because `SpeciesToNationalPokedexNum`allways only returns base form national dex numbers, so we loose the form info

AsparagusEduardo commented on src/pokedex.c:
```diff
@@ -4249,6 +4249,28 @@ u16 GetPokedexHeightWeight(u16 dexNum, u8 data)
+    {
+        return GetPokedexHeightWeight(SpeciesToNationalPokedexNum(species), data);
+    }
+    else if (species >= SPECIES_VENUSAUR_MEGA && species < NUM_SPECIES)
```
Should be `FORMS_START + 1`

TheXaman replied:
alright, changed it

TheXaman commented:
- Added data for ALL forms, if they differ in any way I added them to `gPokedexEntriesForms[species]`, otherwise they are skipped with `PokedexEntryFormSkip[species]` and the base form data is used.
- If the weight and height of a form are the same as the base form, their value is set to 0 and the base form data is used.
- Added categories for all forms, if the category is the same as the base form the value is set as `_("")` and could be skipped with a `==` check (not implemented because the base pokedex cant take advantage)
- Category data is based on https://bulbapedia.bulbagarden.net/wiki/Pok%C3%A9mon_category

AsparagusEduardo commented:
Now that there's a check for having height/weight being 0 to use the base species data, it's redundant to have `PokedexEntryFormSkip`,

TheXaman commented:
> Now that there's a check for having height/weight being 0 to use the base species data, it's redundant to have `PokedexEntryFormSkip`,

Is it? There are still are forms that just have differing pokedex descriptions and those that are 100% the same as the base. The 100% same ones are beeing skipped with the `PokedexEntryFormSkip`, Im not sure how we would differentiate between those and the pokedex only ones

TheXaman commented:
Added Alolan form data:
![pokeemerald-0](https://user-images.githubusercontent.com/48356183/148644821-e358a82d-1433-48fe-baa0-6ddc1933c955.png) ![pokeemerald-1](https://user-images.githubusercontent.com/48356183/148644822-578c14fb-eb26-4dc6-9ba5-266eeef5f047.png)

![pokeemerald-2](https://user-images.githubusercontent.com/48356183/148644831-ce80d363-b88a-4fac-a9ee-c9095806ea60.png)

![pokeemerald-3](https://user-images.githubusercontent.com/48356183/148644834-2ab5dba3-ac2c-4261-aca8-3e377324f0e0.png) ![pokeemerald-4](https://user-images.githubusercontent.com/48356183/148644836-80ba3099-d908-456f-b4e8-9a36585ad396.png)

![pokeemerald-5](https://user-images.githubusercontent.com/48356183/148644840-87773177-44cc-4578-bb37-91ca40b9ea30.png) ![pokeemerald-6](https://user-images.githubusercontent.com/48356183/148644841-221f89f8-7c8d-4e02-af30-acf2ff2ad9cf.png)

![pokeemerald-7](https://user-images.githubusercontent.com/48356183/148644843-4c2c13db-987c-4d42-b843-1f5760760ecf.png) ![pokeemerald-8](https://user-images.githubusercontent.com/48356183/148644844-38fe8138-313d-4725-ac7f-4208d2630da8.png)

![pokeemerald-9](https://user-images.githubusercontent.com/48356183/148644849-debf4616-f861-4b22-a6f6-8b05b6a064ac.png) ![pokeemerald-10](https://user-images.githubusercontent.com/48356183/148644852-71a49fdc-b790-4de2-89c3-45a1bd779a52.png)

![pokeemerald-11](https://user-images.githubusercontent.com/48356183/148644855-5b9fa085-c892-4809-b373-4416384cf9bf.png) ![pokeemerald-12](https://user-images.githubusercontent.com/48356183/148644856-1a4c803c-5340-4a8f-b353-a20cb8c0b7c3.png) ![pokeemerald-13](https://user-images.githubusercontent.com/48356183/148644857-c4c96ac2-b331-48d9-a562-c991b49669dd.png)

![pokeemerald-14](https://user-images.githubusercontent.com/48356183/148644863-f107a265-9086-4dad-a3cd-668a6c7c41c3.png) ![pokeemerald-15](https://user-images.githubusercontent.com/48356183/148644864-240de56a-3b33-4cdf-a60d-51acc19b89fc.png)

![pokeemerald-16](https://user-images.githubusercontent.com/48356183/148644868-c2da5841-470c-4c79-b255-39c4224f34fe.png)

![pokeemerald-17](https://user-images.githubusercontent.com/48356183/148644874-1a3a7948-30e4-422d-8103-d742029b3868.png)

LOuroboros commented on src/data/pokemon/pokedex_text.h:
```diff
@@ -5405,24 +5405,96 @@ const u8 gKyogrePrimalPokedexText[] = _("");
+const u8 gPersianAlolanPokedexText[] = _(
+    "It looks down on everyone other than\n"
+    "itself. In contrast to its lovely face, it\n"
+    "is so brutal that it tortures its weakened\n"
+    "prey rather than finishing them off.");
+const u8 gGeodudeAlolanPokedexText[] = _(
+    "Geodude compete against each other\n"
+    "with headbutts. The iron sand on\n"
+    "their heads will stick to whichever one\n"
+    "has stronger magnetism.");
+const u8 gGravelerAlolanPokedexText[] = _(
+    "They eat rocks and often get\n"
+    "into a scrap over them. The shock of\n"
+    "Graveler smashing together causes\n"
+    "a flash of light and a booming noise.");
+const u8 gGolemAlolanPokedexText[] = _(
+    "It fires rocks charged with electricity.\n"
+    "Even if the rock isn't fired that\n"
+    "accurately, just grazing an opponent\n"
+    "will cause numbness and fainting.");
+const u8 gGrimerAlolanPokedexText[] = _(
+    "There are a hundred or so of them living\n"
+    "in Alola's waste-disposal site. They're all\n"
+    "hard workers who eat a lot of trash. Grimer\n"
+    "seems to relish any and all kinds of trash.");
+const u8 gMukAlolanPokedexText[] = _(
+    "The garbage it eats causes\n"
+    "continuous chemical changes in it's\n"
+    "body, which produce it's\n"
+    "exceedingly vivid coloration.");
+const u8 gExeggutorAlolanPokedexText[] = _(
+    "Alola is the best environment for\n"
+    "this Pokémon. Local people take pride\n"
+    "in its appearance, saying this is how\n"
+    "Exeggutor ought to look.");
+const u8 gMarowakAlolanPokedexText[] = _(
+    "The cursed flames that light up the bone\n"
+    "carried by this Pokémon are said\n"
+    "to cause both mental and physical\n"
+    "pain that will never fade.");
```
```suggestion
const u8 gRattataAlolanPokedexText[] = _(
    "With its incisors, it gnaws through doors\n"
    "and infiltrates people's homes.\n"
    "Then, with a twitch of its whiskers,\n"
    "it steals whatever food it finds.");

const u8 gRaticateAlolanPokedexText[] = _(
    "It forms a group of Rattata, which it \n"
    "assumes command of. Each group\n"
    "has its own territory, and disputes\n"
    "over food happen often.");

const u8 gRaichuAlolanPokedexText[] = _(
    "It uses psychokinesis to control\n"
    "electricity. It focuses psychic energy\n"
    "into its tail and rides it like it's surfing.\n"
    "Another name for this Pokémon is 'hodad'.");

const u8 gSandshrewAlolanPokedexText[] = _(
    "Life on mountains covered with deep snow\n"
    "has granted this Pokémon a body of ice\n"
    "that's as hard as steel.\n"
    "Predators go after its soft belly.");
const u8 gSandslashAlolanPokedexText[] = _(
    "This Pokémon's steel spikes are\n"
    "sheathed in ice. Stabs from these\n"
    "spikes cause wounds and.\n"
    "severe frostbite as well.");

const u8 gVulpixAlolanPokedexText[] = _(
    "They live together in a skulk, helping\n"
    "one another. In hot weather, this Pokémon\n"
    "makes ice shards with its six tails and\n"
    "sprays them around to cool itself off.");

const u8 gNinetalesAlolanPokedexText[] = _(
    "While it will guide travelers who get lost\n"
    "on a snowy mountain down to the\n"
    "mountain's base, it won't forgive anyone\n"
    "who harms nature.");

const u8 gDiglettAlolanPokedexText[] = _(
    "Its head sports an altered form of\n"
    "whiskers made of metal. When in\n"
    "communicationwith its comrades,\n"
    "its whiskers wobble to and fro.");

const u8 gDugtrioAlolanPokedexText[] = _(
    "Their beautiful, metallic whiskers create\n"
    "a sort of protective helmet on\n"
    "heir heads, and they also function\n"
    "as highly precise sensors.");

const u8 gMeowthAlolanPokedexText[] = _(
    "Deeply proud and keenly smart,\n"
    "this Pokémon moves with cunning\n"
    "during battle and relentlessly\n"
    "attacks enemies' weak points.");

const u8 gPersianAlolanPokedexText[] = _(
    "It looks down on everyone other than\n"
    "itself. In contrast to its lovely face, it\n"
    "is so brutal that it tortures its weakened\n"
    "prey rather than finishing them off.");

const u8 gGeodudeAlolanPokedexText[] = _(
    "Geodude compete against each other\n"
    "with headbutts. The iron sand on\n"
    "their heads will stick to whichever one\n"
    "has stronger magnetism.");

const u8 gGravelerAlolanPokedexText[] = _(
    "They eat rocks and often get\n"
    "into a scrap over them. The shock of\n"
    "Graveler smashing together causes\n"
    "a flash of light and a booming noise.");

const u8 gGolemAlolanPokedexText[] = _(
    "It fires rocks charged with electricity.\n"
    "Even if the rock isn't fired that\n"
    "accurately, just grazing an opponent\n"
    "will cause numbness and fainting.");

const u8 gGrimerAlolanPokedexText[] = _(
    "There are a hundred or so of them living\n"
    "in Alola's waste-disposal site. They're all\n"
    "hard workers who eat a lot of trash. Grimer\n"
    "seems to relish any and all kinds of trash.");

const u8 gMukAlolanPokedexText[] = _(
    "The garbage it eats causes\n"
    "continuous chemical changes in it's\n"
    "body, which produce it's\n"
    "exceedingly vivid coloration.");

const u8 gExeggutorAlolanPokedexText[] = _(
    "Alola is the best environment for\n"
    "this Pokémon. Local people take pride\n"
    "in its appearance, saying this is how\n"
    "Exeggutor ought to look.");

const u8 gMarowakAlolanPokedexText[] = _(
    "The cursed flames that light up the bone\n"
    "carried by this Pokémon are said\n"
    "to cause both mental and physical\n"
    "pain that will never fade.");
```

AsparagusEduardo commented:
Superseded by #2363
